cmd/skywire-cli/commands/visor/logserver.go

```diff
@@ -0,0 +1,98 @@
+package clivisor
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/skycoin/dmsg/pkg/disc"
+	dmsg "github.com/skycoin/dmsg/pkg/dmsg"
+	"github.com/spf13/cobra"
+
+	"github.com/skycoin/skywire-utilities/pkg/cipher"
+	"github.com/skycoin/skywire-utilities/pkg/cmdutil"
+	"github.com/skycoin/skywire-utilities/pkg/logging"
+	"github.com/skycoin/skywire/pkg/skyenv"
+	"github.com/skycoin/skywire/pkg/visor/visorconfig"
+)
+
+var (
+	dir            = skyenv.PackageConfig().LocalPath // local dir to serve via http
+	dmsgDisc       = "http://dmsgd.skywire.skycoin.com"
+	dmsgPort       = uint(81)
+	pubkey, seckey = cipher.GenerateKeyPair() //nolint
+)
+
+func init() {
+	RootCmd.AddCommand(logserverCmd)
+	logserverCmd.Flags().SortFlags = false
+	logserverCmd.Flags().StringVarP(&dir, "dir", "d", dir, "local dir to serve via http")
+	logserverCmd.Flags().StringVarP(&dmsgDisc, "disc", "e", dmsgDisc, "dmsg discovery address")
+	logserverCmd.Flags().UintVarP(&dmsgPort, "port", "p", dmsgPort, "dmsg port to serve from")
+	logserverCmd.Flags().Var(&seckey, "sk", "dmsg secret key")
+	logserverCmd.Flags().MarkHidden("sk") //nolint
+}
+
+var logserverCmd = &cobra.Command{
+	Use:   "logserver",
+	Short: "log server",
+	Long: `dmsghttp log server
+
```

Comment:
Initially, I started with copying the example dmsghttp server from the dmsg repo into a cli command, with the intent of running it as its own service independent of the visor.

I realized this would not work because we would need a similar non-systemd framework for this service on windows and macOS as we want to collect logs for transports from vpn clients as well as servers for transport log reconciliation.

So I wrote the logserver into the visor, but then removed it as it was not working correctly and erson had it working in his pr.

logserver.go is the initial effort in this direction which was ultimately abandoned and will be removed with the next commit